OSDOCS-3984: Port Authentication and authorization book to OSD and ROSA (#67047)
openshift/openshift-docs: authentication/index.adoc

> 

Reviewer: Should the glossary page go last? It's currently showing up first.

[id="authentication-overview"]
== About authentication in {product-title}
To control access to an {product-title} cluster,
ifndef::openshift-dedicated,openshift-rosa[]
a cluster administrator
endif::openshift-dedicated,openshift-rosa[]
ifdef::openshift-dedicated,openshift-rosa[]

> an administrator with the `dedicated-admin` role

Reviewer: ```suggestion
an administrator with the `dedicated-admin` or `cluster-admin` role
```

Author: While this is true, the OSD and ROSA docs target the `dedicated-admin` in the sections that follow.

endif::openshift-dedicated,openshift-rosa[]
can configure xref:../authentication/understanding-authentication.adoc#understanding-authentication[user authentication] and ensure only approved users access the cluster.

To interact with an {product-title} cluster, users must first authenticate to the {product-title} API in some way. You can authenticate by providing an xref:../authentication/understanding-authentication.adoc#rbac-api-authentication_understanding-authentication[OAuth access token or an X.509 client certificate] in your requests to the {product-title} API.

[NOTE]
====
If you do not present a valid access token or certificate, your request is unauthenticated and you receive an HTTP 401 error.
====

ifdef::openshift-dedicated,openshift-rosa[]

> An administrator can configure authentication by configuring an identity provider. You can define any xref:../authentication/sd-configuring-identity-providers.adoc#understanding-idp-supported_sd-configuring-identity-providers[supported identity provider in {product-title}] and add it to your cluster.

Reviewer 2: ```suggestion
An administrator can set up authentication by configuring an identity provider. You can define any xref:../authentication/sd-configuring-identity-providers.adoc#understanding-idp-supported_sd-configuring-identity-providers[supported identity provider in {product-title}] to integrate with your cluster
```

Author: I think "configure" works better here. "Set up" implies something that you might do once, whereas adding IDP providers is something you can do anytime.

endif::openshift-dedicated,openshift-rosa[]

ifndef::openshift-dedicated,openshift-rosa[]
An administrator can configure authentication through the following tasks:

* Configuring an identity provider: You can define any xref:../authentication/understanding-identity-provider.adoc#supported-identity-providers[supported identity provider in {product-title}] and add it to your cluster.

> * xref:../authentication/configuring-internal-oauth.adoc#configuring-internal-oauth[Configuring the internal OAuth server]: The {product-title} control plane includes a built-in OAuth server that determines the user's identity from the configured identity provider and creates an access token. You can configure the token duration and inactivity timeout, and customize the internal OAuth server URL.

Reviewer 2: How about below 

```suggestion
* xref:../authentication/configuring-internal-oauth.adoc#configuring-internal-oauth[Configuring the internal OAuth server]: The {product-title} control plane comes with a built-in OAuth server. This server identifies the user based on the configured identity provider and generates an access token. You have the option to configure the token's duration, set an inactivity timeout, and customize the internal OAuth server URL.
```

Author: This content only applies to OCP, so I'll leave it up to them if they'd like to revise this.
